### ISSUE TITLE
bug: eza reports broken symlinks incorrectly

### ISSUE THREAD

**Issue by alexpyattaev** [eza-community/eza#255]
When listing a directory with broken symlinks, eza does strange stuff and tries to follow them. This causes it to report OS error. Detailed log/example below. 

- The version of eza being used (`eza --version`) v0.11.1 [+git]
- The command-line arguments you are using: any arguments will trigger the issue
- Your operating system and hardware platform Arch, x86_64

---
What actually happens:
```
tmp ❯ ls -l
[./qwerwqerqew: No such file or directory (os error 2)]
[./rqwerwe: No such file or directory (os error 2)]
srwxr-xr-x   0 USERNAME 11 Sep 18:24 cantata-tags-df59826
.rw-r--r-- 106 USERNAME 11 Sep 18:26 f79d601e26a782fd149b3ffb098aae9f
drwx------   - root       11 Sep 18:35 mc-root
```

**Comment by cafkafk** [comment]
This might be related to #112

**Comment by Stetsed** [comment]
I am experiencing the same issue on 0.11.1

**Comment by cafkafk** [comment]
We are looking for windows users that can test the fix in #262, that's all we need to merge and solve this

**Comment by cafkafk** [comment]
> > We are looking for windows users that can test the fix in #262, that's all we need to merge and solve this
> 
> I have a Windows Box, anything specific I need to test?

@cfxegbert might know more, but just try it out and see if there are any obvious bugs, run it on some broken symlinks and whatever else you can come up with really.

**Comment by Stetsed** [comment]
I'm sorry but it seems like I am lacking braincells today(and I haven't used Windows in a long ass while) so I can't even get it to compile because of compilers missing and for the life of me I can't figure it out. Apologies and I hope somebody else can test it so a fix can be pushed out.
